packages/core/src/stateUtils.ts

```diff
@@ -929,15 +929,20 @@ function getPathFromRootToNode<
   TC extends MachineContext,
   TE extends EventObject
 >(stateNode: StateNode<TC, TE>): Array<StateNode<TC, TE>> {
-  const path: Array<StateNode<TC, TE>> = [];
   let marker = stateNode.parent;
 
+  if (!marker) {
+    return [stateNode];
+  }
+
+  const path: Array<typeof stateNode> = [];
+
   while (marker) {
-    path.unshift(marker);
+    path.push(marker);
     marker = marker.parent;
   }
 
-  return path;
+  return path.reverse();
```

Comment:
Conceptually it's a little bit better to push to an array than to shift it all the time (each shift has to move all the existing items). So I've switched to pushing here but to correct the actual result I need to `.reverse()` at the end.